observer-errors/README.md

```diff
@@ -0,0 +1,130 @@
+
+While the `.onerror` remains the classic approach for handling errors in window and elements, it doesn't fit well in the Observer the same way it can't add an event listener as in `window.addEventListener`. This matches the idea on [observers API not being based in events](https://github.com/WICG/resize-observer/blob/master/explainer.md#why-an-observer-based-api-and-not-events).
+
+This proposal tries a less complex solution that uses another callback for catching the errors.
+​
+```js
+var observer = new ResizeObserver(function(entries) { /* ... */ });
+observer.observe(elem1);
+observer.observe(elem2);
+observer.observe(elem3);
+
+// The new proposed method
+observer.catch(function(err) {
+    /* error to be captured when the callback is executed */
+});
+```
+
+This `catch` name is heavily inspired on the Promise's API but naming can be discussed and the current choice is not a hard requirement.
+
+It's important to observe that registering the `.catch` callback can remove the bubbling out. It may work as an opt-in to ignore errors:
+
+```js
+observer.catch(function(err) { /* noop */ });
+```
+
+This method captures errors triggered in the callback function. E.g.:
+
+```js
+var observer = new ResizeObserver(function(entries) {
+    throw new Error('meep');
+});
+observer.observe(elem1);
+observer.observe(elem2);
+
+// The new proposed method
+observer.catch(function(err) {
+    console.log(err.message); // 'meep'
+});
+```
+
+### Open ended discussions and Variants
+
+#### Multiple catch calls
+
+What it means to call `.catch` more than once? 
+
+```js
+observer.catch(function(err) { console.log('1'); });
+observer.catch(function(err) { console.log('2'); });
+```
+
+The current suggestion is for the observer to register every callback, instead of replacing them. This remains [similar to the Promise's `catch`](https://jsfiddle.net/2mpzoLrq/). The error object would be preserved for each call.
+
+#### No definition (yet) to unregister catch callbacks.
+
+This explainer does not define any way to unregister catch callbacks as there is little to no identified value in adding this so far.
+
+#### Capturing the effects from observed elements
+
+The catch callback can be used to capture errors triggered in the Observer callback but also expanded to error handling as described in the [`Effects within the observable elements`](#effects) section.
+
+#### Setting the error callback in the constructor API
+
+An alternative for the `.catch` method would be setting the error callback in the Observer constructor. E.g.:
+
+```js
+function fn(entries) { /* noop */ }
+function errCallbackFn(err) { /* noop */ }
+
+var observer = new ResizeObserver(fn, errCallbackFn);
+```
+
+Although, this would only allow registering one error callback and it gives bad intuition the error callback can be used the error is triggered by calling the regular callback function `fn`.
```

Comment:
I don't quite understand this sentence... can it be broken up / reworded?

Another argument for why this would be a bad pattern: it's the exact opposite of the Node ["errback" pattern](http://fredkschott.com/post/2014/03/understanding-error-first-callbacks-in-node-js/). (E.g. Node's `fs` module – `fs.readFile('foo.txt', (err, data) => {})`.)

Reply:
I'm rewording this, but this is not really like the errback pattern. There are still a main callback (first parameter) and a separate error callback. They are separate and yes, it's just not good.

Perhaps I should just remove this section and remove it as an alternative.